packages/parser/src/parse.ts

```diff
@@ -0,0 +1,42 @@
+import type { LangiumParser, ParseResult } from 'langium';
+
+import type { Info } from './index.js';
+import { createInfoServices } from './language/index.js';
+
+export type DiagramAST = Info;
+
+const parsers: Record<string, LangiumParser> = {};
+
+const initializers = {
+  info: () => {
+    // Will have to make parse async to use this. Can try later...
+    // const { createInfoServices } = await import('./language/info/index.js');
+    const parser = createInfoServices().Info.parser.LangiumParser;
+    parsers['info'] = parser;
+  },
+} as const;
+
+export const parse = <T extends DiagramAST>(
+  diagramType: keyof typeof initializers,
+  text: string
+): T => {
+  const initializer = initializers[diagramType];
+  if (!initializer) {
+    throw new Error(`Unknown diagram type: ${diagramType}`);
+  }
+  if (!parsers[diagramType]) {
+    initializer();
+  }
+  const parser: LangiumParser = parsers[diagramType];
+  const result: ParseResult<T> = parser.parse<T>(text);
+  if (result.lexerErrors.length > 0 || result.parserErrors.length > 0) {
+    throw new MermaidParseError(result);
+  }
+  return result.value;
+};
+
+export class MermaidParseError extends Error {
+  constructor({ lexerErrors, parserErrors }: ParseResult<DiagramAST>) {
+    super(`Parsing failed: ${lexerErrors} ${parserErrors}`);
```

Comment:
```suggestion
  constructor(public result: ParseResult<DiagramAST>) {
    super(`Parsing failed: ${result.lexerErrors} ${result.parserErrors}`);
```

How does `Parsing failed: ${result.lexerErrors} ${result.parserErrors}` show up in the console?
It's useless if it's just `[Object object]`

Reply:
```
12.447 : ERROR :  Error parsing Error: Parsing failed: [object Object] 
```

Reply:
Should we `JSON.stringify`?

Reply:
We could use it. Or, we could use something like `lexerErrors.map(err => err.message).join("\n")`, so it's rendered nicely.